### ISSUE TITLE
pre_capture is obeyed but post_capture is not obeyed

### ISSUE THREAD

**Issue by gtsiros** [Motion-Project/motion#191]
Running version  3.2.12+git20140228 on a Rpi 3 B with a microsoft hd3000 usb webcam

motion.conf:

daemon on
setup_mode off

v4l2_palette 17
width 1280
height 720
framerate 4
mask_file /home/pi/mask.pgm
despeckle_filter EedDl
pre_capture 20
post_capture 20
event_gap 30
output_pictures on
quality 75
stream_port 20043
stream_quality 25
stream_maxrate 2
stream_localhost off
on_motion_detected play -q -n synth 0.2 sq 880
threshold 1500
webcontrol_port 20044
webcontrol_localhost off
target_dir /mnt/rpi/captures
text_double on
text_changes on
locate_motion_mode on

everything else is whatever is the default

it saves the 20 images before motion is detected, but no more than that

I checked it my putting my hand in front of the camera, the last image is the one in which motion has been detected

Any ideas?

edit: I'm new to this whole bug reporting thing, if i should have mentioned something more just tell me and I will try to provide any information. So far I wrote all that i thought relevant.

**Comment by Mr-Dave** [comment]
We will need you to upgrade your version and validate the issue occurs with something more current.  The releases page [here](https://github.com/Motion-Project/motion/releases) has deb packages that will get something more current.  You should be able to install the appropriate one via the gdebi tool

**Comment by gtsiros** [comment]
exact same behavior.

also, decreased:
resolution from 1280x720 to 320x240
framerate from 4 to 2
pre_capture from 20 to 4
post_capture from 20 to 8
webcam max framerate to 1
removed on_motion_action in case sox uses too much cpu

again, exact same behavior. It does not obey post_capture at all

**Comment by pcwalden** [comment]
I have noticed this behaviour as well for some time. Back to the pre-Motion-Project days.

The post_capture configuration attribute never seems to be honoured. The frame is as if post_capture = 0.

This is with netcams only. I cannot say for sure about MMAL cameras on Pi's.

**Comment by gtsiros** [comment]
Maybe it works only if **motion** is set up to save movies, not just still images?

could you share your settings via copy-paste from the web-interface which shows all variables?

**Comment by pcwalden** [comment]
Here is my camera 1 config list. post_capture = 3. Acts like post_capture = 0

```
Thread 1

    setup_mode = off
    camera_name = (not defined)
    videodevice = /dev/video0
    v4l2_palette = 17
    input = -1
    norm = 0
    frequency = 0
    rotate = 0
    width = 320
    height = 240
    framerate = 2
    minimum_frame_time = 0
    netcam_url = http://192.168.1.150:8081/image/jpeg.cgi
    netcam_userpass = xxxxx:yyyyzzzz
    netcam_proxy = (not defined)
    netcam_tolerant_check = off
    auto_brightness = off
    brightness = 0
    contrast = 0
    saturation = 0
    hue = 0
    power_line_frequency = -1
    roundrobin_frames = 1
    roundrobin_skip = 1
    switchfilter = off
    threshold = 200
    threshold_tune = on
    noise_level = 32
    noise_tune = on
    despeckle_filter = EedDl
    area_detect = (not defined)
    mask_file = /mnt/webcam/cam1mask.pgm
    smart_mask_speed = 0
    lightswitch = 10
    minimum_motion_frames = 2
    pre_capture = 1
    post_capture = 3
    event_gap = 60
    max_movie_time = 0
    emulate_motion = off
    output_pictures = on
    output_debug_pictures = off
    quality = 90
    picture_type = jpeg
    ffmpeg_output_movies = off
    ffmpeg_output_debug_movies = off
    ffmpeg_timelapse = 0
    ffmpeg_timelapse_mode = daily
    ffmpeg_bps = 400000
    ffmpeg_variable_bitrate = 0
    ffmpeg_video_codec = mpeg4
    ffmpeg_duplicate_frames = off
    use_extpipe = off
    extpipe = (not defined)
    snapshot_interval = 0
    locate_motion_mode = off
    locate_motion_style = box
    text_right = %Y-%m-%d\n%T-%q
    text_left = (not defined)
    text_changes = off
    text_event = %Y%m%d%H%M%S
    text_double = off
    exif_text = (not defined)
    target_dir = /mnt/webcam/cam1
    snapshot_filename = %Y/%m/%d/%H%M%S-%q
    picture_filename = %Y/%m/%d/%H%M%S-%q
    movie_filename = %v-%Y%m%d%H%M%S
    timelapse_filename = %Y%m%d-timelapse
    ipv6_enabled = off
    stream_port = 0
    stream_quality = 90
    stream_motion = off
    stream_maxrate = 1
    stream_localhost = on
    stream_limit = 0
    stream_auth_method = 0
    stream_preview_scale = 25
    stream_preview_newline = off
    track_type = 0
    track_auto = off
    track_port = (not defined)
    track_motorx = 0
    track_motorx_reverse = off
    track_motory = 0
    track_motory_reverse = off
    track_maxx = 0
    track_minx = 0
    track_maxy = 0
    track_miny = 0
    track_homex = 128
    track_homey = 128
    track_iomojo_id = 0
    track_step_angle_x = 10
    track_step_angle_y = 10
    track_move_wait = 10
    track_speed = 255
    track_stepsize = 40
    quiet = on
    on_event_start = (not defined)
    on_event_end = (not defined)
    on_picture_save = (not defined)
    on_motion_detected = (not defined)
    on_area_detected = (not defined)
    on_movie_start = (not defined)
    on_movie_end = (not defined)
    on_camera_lost = (not defined)
    video_pipe = (not defined)
    motion_video_pipe = (not defined)

```

**Comment by tosiara** [comment]
Has anyone tried to reproduce it with the latest 4.x release?

**Comment by bcl** [comment]
I've been noticing this as well. I post-process the saved images using another program so I was expecting post_capture to save the images. Ends up this only happens if ffmpeg_ouput or use_extpipe+extpipe are used. If you are only saving images they aren't saves. I've submitted pull #284 to change this behavior and always save them which I think is more consistent with the expected behavior.

**Comment by Mr-Dave** [comment]
Closed via pull request #284
